timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/NamespaceTakeoverService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.timelock.paxos.api.UndertowNamespaceLeadershipTakeoverService;
+import com.palantir.tokens.auth.AuthHeader;
+import java.util.Set;
+
+final class NamespaceTakeoverService implements UndertowNamespaceLeadershipTakeoverService {
```

Comment:
These endpoints were added for timelock partitioning/timelock batching (discussed in our sync with jkong) close to the end of that work (PR 5X based on titles). I found no usage from metrics.

All conjure defined services have async interfaces and someone in the past just extracted this to a common conjure.gradle file. Making conjure generate a synchronous interface would either require hackiness or require making this subproject non standard. Also, unfortunately conjure only supports opt-in async if the global default is sync (and the other way round).

This call seems to be making some external requests at some point so might have been okay to truly make the impl async (see conjure or conjure-java readme, async is typically discouraged except for some cases, an example is delegating to an async dialogue call) but that would require deep plumbing and making the whole calling chain async.

Based off all of this, async-as-sync is fine. I also tested this on IL on some example calls for each endpoint and all worked.